app/controllers/hiring_staff/vacancies/job_specification_controller.rb

```diff
@@ -32,6 +32,7 @@ def create
 
   def update
     if @job_specification_form.complete_and_valid?
+      remove_subject_fields(@vacancy)
```

Comment:
This is a nice place for this method. I'm not sure how we know that the rake task for converting legacy subject fields to `subjects` array is run before this method is run, and so how we know that we aren't losing the legacy subject fields.